TruePath/PathStrings.cs

```diff
@@ -36,61 +37,93 @@ public static class PathStrings
     ///     Note that this operation will never perform any file IO, and is purely string manipulation.
     /// </para>
     /// </summary>
+    [SkipLocalsInit] // is necessary to prevent the CLR from filling stackalloc with zeros.
     public static string Normalize(string path)
     {
-        // TODO[#19]: Optimize this. It is possible to do with less allocations.
-        var segments = new List<(int Start, int End)>();
+        int written = 0;
+        Span<char> normalized = stackalloc char[path.Length];
+        ReadOnlySpan<char> source = path.AsSpan();
```

Comment:
There's a problem with this approach.

My plan is to not have any path length limits in the future. For most normal file systems, the limit is 4k or lower, but
- there may be cases when it is a bit longer (for a relative and/or denormalized path)
- for a virtual file system, there is no practical limit

If somebody passes a path of 1 MiB to here, this code may easily fail and crash the whole process, which is not acceptable for a good-behaving library.

After some thought, I decided to use the following approach:
- if the path fits into 4k chars, let's use `stackalloc`
- otherwise, let's rent a heap-allocated array from `ArrayPool.Shared`

This should not hurt the common cases when the paths have normal length, and should even work well enough for cases when the paths are overly long.